controllers/flp/flp_controller.go

```diff
@@ -253,3 +266,84 @@ func reconcileLokiRoles(ctx context.Context, r *reconcilers.Common, b *builder)
 	}
 	return nil
 }
+
+func (r *Reconciler) getOpenShiftSubnets(ctx context.Context) ([]flowslatest.SubnetLabel, error) {
+	var subnets []flowslatest.SubnetLabel
+
+	// Pods and Services subnets are found in CNO config
+	if r.mgr.HasCNO() {
+		network := &operatorv1.Network{}
+		err := r.Get(ctx, types.NamespacedName{Name: "cluster"}, network)
+		if err != nil {
+			return nil, fmt.Errorf("can't get Network information: %w", err)
+		}
+		var podCIDRs []string
+		for _, podsNet := range network.Spec.ClusterNetwork {
+			podCIDRs = append(podCIDRs, podsNet.CIDR)
+		}
+		if len(podCIDRs) > 0 {
+			subnets = append(subnets, flowslatest.SubnetLabel{
+				Name:  "Pods",
+				CIDRs: podCIDRs,
+			})
+		}
+		svcCIDRs := network.Spec.ServiceNetwork
+		if len(svcCIDRs) > 0 {
+			subnets = append(subnets, flowslatest.SubnetLabel{
+				Name:  "Services",
+				CIDRs: svcCIDRs,
+			})
+		}
+	}
+
+	// Nodes subnet found in CM cluster-config-v1 (kube-system)
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: "cluster-config-v1", Namespace: "kube-system"}, cm); err != nil {
+		return nil, fmt.Errorf(`can't read "cluster-config-v1" ConfigMap: %w`, err)
+	}
+	machines, err := readMachineNetworks(cm)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
so yes, that's right, but for some reason I feel the CNO would be a more reliable source .. tbh this is just a gut feeling. Relying on the install config map sounds more shaky, no? First because it's not a CRD but a ConfigMap so it's not backed by a stable go API for unmarshalling. Second, because I am not 100% sure if every installation method deploys this configmap, and also I wonder if it's possible to remove it after the install.

Reply:
since u already marshaling the cm to get machineNetwork so I felt it makes sense to pull clusterNetwork from there as well, I am this cm is there for all install but worth investigating as this will simplify the logic here IMO

Reply:
Finally got an answer from installer folks about that, they say it's more or less like I suspected, they don't encourage relying on this CM and rather use the other APIs when possible - but for machineNetwork in particular that's apparently the best place to look ; so I think I'll keep using this pattern, using `Network` api where I can, and fallback on the CM otherwise.